src/plugins/home/server/services/sample_data/routes/install.ts

```diff
@@ -165,10 +177,27 @@ export function createInstallRoute(
         }
       }
 
+      let dataSourceTitle;
+      if (dataSourceId) {
+        const dataSource = await context.core.savedObjects.client
+          .get('data-source', dataSourceId)
```

Comment:
let's add a try catch here in case the get data source operation fails.

shall we do this before inserting data into remote data source? and fail this API call if the get data source fails